### ISSUE TITLE
feat: ionic-vue ion-tab-button be able to use replace

### ISSUE THREAD

**Issue by scottix** [ionic-team/ionic-framework#20084]
# Feature Request
ionic-vue feature request to use :replace instead of :to to prevent pages from being pushed on to the navigation stack.

**Ionic version:**
[x] **4.x**

**Describe the Feature Request**

I want to be able to switch through the tabs and not have to push a navigation history on each item. Because with Android back button it will go through your entire history of switching tabs, which is un-desirable.

**Describe Preferred Solution**

It would be nice to just have something like this.

```
<ion-tab-button tab="someTab" :replace="{name: 'someTab'}>
</ion-tab-button>
```

**Describe Alternatives**

Been using this alternative
```
<ion-tab-button tab="someTab" @click="$router.replace({name: 'someTab'})>
</ion-tab-button>
```

**Related Code**
https://github.com/ionic-team/ionic/blob/master/vue/src/components/navigation/ion-tabs.ts#L145

**Additional Context**

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you please describe your use case for using `replace` on an `ion-tab-button`? The current behavior is what aligns with native tabs applications.

**Comment by scottix** [comment]
Ya this was a while ago when we were using the previous beta, for our use case we were trying to avoid the back navigation when switching between tabs.

An example would be and to reference my sample project. https://github.com/scottix/vue-pretab

Navigation order:
1. Pretab
2. Tab1
3. Tab3
4. Tab2
5. Tab1

The back navigation if used would follow backwards in that order.
1. Tab2
2. Tab3
3. Tab1
4. Pretab

Basically we don't care about what order of tabs you went to, we basically wanted from anypoint always take you to the Pretab. Now maybe things are different I haven't fully looked at what we could do.

**Comment by liamdebeasi** [comment]
Hi there,

Thanks for filing this issue and providing more context around your use case. The team has decided not to implement this feature.

Tabs in mobile applications are intended to behave as individual stacks. This behavior is modeled after tabs in native mobile applications. In this scenario, items in one stack never interact with items in another stack. In other words, Tab 2  can never influence the state of Tab 3. The proposed feature would go against this design as it means the state of the previous tab would get replaced with the state of the current tab.

If this is a behavior you need for your app, we recommend not using the built-in tabs. You can use a nested `ion-router-outlet` with `ion-tab-bar` (or a custom component) which should allow you to get the behavior listed on this thread.

I am going to close this. Thanks!
